app/fissile.go

```diff
@@ -842,3 +877,255 @@ roleLoop:
+	// map[string]string
+
+	properties := manifestProperties(roleManifest)
+	// map[string]string
+
+	// All properties must be defined in a BOSH release
+	allErrs = append(allErrs, validateBosh("role-manifest",
+		properties, bosh)...)
+
+	// All light opinions must exists in a bosh release
+	allErrs = append(allErrs, validateBosh("light opinion",
+		light, bosh)...)
+
+	// All dark opinions must exists in a bosh release
+	allErrs = append(allErrs, validateBosh("dark opinion",
+		dark, bosh)...)
+
+	// All dark opinions must be configured as templates
+	allErrs = append(allErrs, darkExposed(dark, properties)...)
+
+	// No dark opinions must have defaults in light opinions
+	allErrs = append(allErrs, darkUnexposed(dark, light)...)
+
+	// No duplicates must exist between role manifest and light
+	// opinions
+	allErrs = append(allErrs, checkOverrides(light, roleManifest)...)
+
+	// All bosh properties in a release should have the same
+	// default across jobs -- WARNING only, not error
+	f.checkBoshDefaults(bosh)
+
+	// All light opinions should differ from their defaults in the
+	// BOSH releases
+	allErrs = append(allErrs, f.checkLightDefaults(light, bosh)...)
+
+	// 	allErrs = append(allErrs, XXX()...)
+
+	return allErrs
+}
+
+func validateBosh(label string, properties map[string]string, bosh propertyDefaults) validation.ErrorList {
+	// All provided properties must be defined in a BOSH release
+	allErrs := validation.ErrorList{}
+
+	for property := range properties {
+		// Ignore specials (without the "properties." prefix)
+		if !strings.HasPrefix(property, "properties.") {
+			continue
+		}
+		p := strings.TrimPrefix(property, "properties.")
+
+		if _, ok := bosh[p]; !ok {
+			allErrs = append(allErrs, validation.NotFound(
+				fmt.Sprintf("%s '%s'", label, p), "In any BOSH release"))
+		}
+	}
+
+	return allErrs
+}
+
+func manifestProperties(roleManifest *model.RoleManifest) map[string]string {
+	properties := make(map[string]string)
+
+	// Per-role properties
+	for _, role := range roleManifest.Roles {
+		for property, template := range role.Configuration.Templates {
+			if _, ok := properties[property]; ok {
+				continue
+			}
+			properties[property] = template
+		}
+	}
+
+	// And the global properties
+	for property, template := range roleManifest.Configuration.Templates {
+		if _, ok := properties[property]; ok {
+			continue
+		}
+		properties[property] = template
+	}
+
+	return properties
+}
+
+func darkExposed(dark map[string]string, properties map[string]string) validation.ErrorList {
+	// All dark opinions must be configured as templates
+	allErrs := validation.ErrorList{}
+
+	for property := range dark {
+		if _, ok := properties[property]; ok {
+			continue
+		}
+		allErrs = append(allErrs, validation.NotFound(
+			property, "Dark opinion is missing template in role-manifest"))
+	}
+
+	return allErrs
+}
+
+func darkUnexposed(dark map[string]string, light map[string]string) validation.ErrorList {
```

Comment:
Used `checkForDarkInTheLight` ;) Commit sometime tomorrow, still working through the list.